src/_macosx.m

```diff
@@ -611,24 +611,25 @@ static CGFloat _get_device_scale(CGContextRef cr)
     View* view;
     PyObject* size;
     int width, height;
-    PyObject* obj;
     FigureCanvas* canvas;
 
     if (!self->window) {
         PyErr_SetString(PyExc_RuntimeError, "NSWindow* is NULL");
         return -1;
     }
+    static char* kwlist[] = { "canvas", NULL };
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!", kwlist,
```

Comment:
Technically this should be in a separate PR?

Reply:
I'm not sure I follow, why did you update this here? Now that you changed back to a method, shouldn't this just be left alone? So, I guess yes, if you want to update the init method it should be a separate PR.

Reply:
There was an acceptance of any Python object with an unsafe cast to a _macos.FigureCanvas. I at least added a type check which should be safe in most scenarios.

I'll put it in a separate PR.